tests/unit/array/storage/elastic/test_del.py

```diff
@@ -47,3 +48,43 @@ def test_delete_offset_success_sync_es_offset_index(deleted_elmnts, start_storag
             index=elastic_doc._index_name_offset2id, query={'match': {'blob': id}}
         )['hits']['hits'][0]['_id']
         assert actual_offset_index == expected_offset
+
+
+@pytest.mark.filterwarnings('ignore::UserWarning')
+def test_success_handle_bulk_delete_not_found(start_storage):
```

Comment:
I would prefer not to test the private methods. Why can't this be tested with the public API?

Reply:
because when I want try to delete non existed ids, the docarray data structure already handle this and raise error before sending bulk request to elastic search. Thus I test using private method. 

The actual error I get on delete operation is actually not on deleting non existed ids (docarray already guard it), but the following error 

```python
{'delete': {'_index': 'offset2id__docarray_unit_apartement__v0_1', '_id': '52', '_version': 4, 'result': 'not_found', '_shards': {'total': 2, 'successful': 1, 'failed': 0}, '_seq_no': 832, '_primary_term': 2, 'status': 404}}
```

this error is because we are using multiple shards, but data is only found on 1 shard. However I dont really know how to reflect this exact detail on your test. Thus I change it to similar scope ( on delete ops, but change the error on not found error ) with the same handling logic